### PR TITLE
SRT refactor

### PR DESCRIPTION
**Pull request by hnyls2002**
This PR is to clarify the logic of the SRT code.

- Move `InputMetaData` and `ForwardMode` into `ForwardInfo`
- Add `PrefillAdder`
- Organize `prepare_for_extend`
- Move some preparation into `InputMetadata`

This PR has been split into #960, #967, #968, #977, #980, #984, #981

### CODE REVIEW

File: python/sglang/srt/models/qwen2_moe.py [hunk 1]
```diff
@@ -368,7 +366,6 @@ def __init__(
             config.vocab_size, config.hidden_size, quant_config=quant_config
         )
         self.logits_processor = LogitsProcessor(config)
-        self.sampler = Sampler()
```
Comment (min-xu-et) [inline python/sglang/srt/models/qwen2_moe.py]: just curious, why is this change part of this PR?

**Review by min-xu-et — COMMENTED** [review]
I am new to the code base. This seems like a code movement change? No logical change other than the removal of "sampler" in qwen model code?

File: python/sglang/srt/models/qwen2_moe.py (hunk 1 above)
Comment (hnyls2002) [reply python/sglang/srt/models/qwen2_moe.py]: We don't use vllm's sampler in our code base, this is a mistake by the qwen's contributor.

**Comment by zhyncs** [comment]
I'll review asap 😂

**Pull request closed without merging**